api/vpc/v1beta1/externalattachment_types.go

```diff
@@ -136,12 +137,19 @@ func (attach *ExternalAttachment) Validate(ctx context.Context, kube kclient.Rea
 	if attach.Spec.Switch.IP == "" {
 		return nil, errors.Errorf("switch.ip is required")
 	}
+	if _, _, err := net.ParseCIDR(attach.Spec.Switch.IP); err != nil {
+		return nil, errors.New("switch.ip is not a valid IP CIDR") //nolint: goerr113
+	}
 	if attach.Spec.Neighbor.ASN == 0 {
 		return nil, errors.Errorf("neighbor.asn is required")
 	}
+	//TODO: do we support 4 octets ASN? else we should check for max 65535
```

Comment:
Yes, we support passing it to the Broadcom SONiC and it seems to be supporting it just fine (I had to make a change in gNMI support code some time ago b/c they've started allowing it in the API)

We've never tested it which is probably a good idea

Reply:
@edipascale could you please create a ticket to check extended ASNs for bpth spine/leaf and externals? thx